### DISCUSSION TITLE
Medium=Events

### DISCUSSION THREAD

**Discussion by samsethi** [Podcastindex-org/podcast-namespace#640] [Tag Proposal]
I would like to propose a new Node extension to the [Medium](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#medium) tag called **Events**.  **#Phase8**

I have been working with a startup in London called [PodLife Events](https://podlifeevents.com/).  We discussed creating an Event Feed with a list of events.  This feed would have a link in the podcasters main RSS Feed, similar to [Podrolls](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#podroll) and [Publisher Feeds](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#publisher).  

@nathangathright then created the example below. 

### Example - Podcast Feed
```
<rss xmlns:podcast="https://podcastindex.org/namespace/1.0"
  xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd"
  xmlns:content="http://purl.org/rss/1.0/modules/content/"
  xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
<channel>
  <title>Nourish Your Network</title>
  <podcast:medium>podcast</podcast:medium>
  <podcast:events>
    <podcast:remoteItem medium="events" title="Nourish Your Network: The Confidence Booster Session" feedGuid="..." feedUrl="..."/>
  </podcast:events>
</channel>
</rss>
```

In TrueFans we built this as a demo.  We built the admin dashboard backend for podcasters to create events. We then show the event(s) on a [podcasters page.](https://truefans.fm/oh-for-foods-sake?tab=events)    

**Image 1:** TrueFans showing the event on the podcasters page. 

<img width="865" alt="TrueFans___Shows___Oh_For_Food_s_Sake" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/a6805888-c9cc-4e4d-a49e-023a718403f6">

In TrueFans if there is no URL in the ticket link field, we enable the fan to pay via Fiat or Sats on platform. This does not have to be part of the spec. 
  
**Image 2:** TrueFans payment for an Event via SATs or Fiat. 

<img width="956" alt="TrueFans___Shows___Oh_For_Food_s_Sake" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/5d5201eb-2a71-4d6d-8cf9-4e215b59c852">

In the admin dashboard we added a title, start and end dates. We added description, ticket price and currency.  

**Image 3.** TrueFans backend for Podcasts to create an Event with tickets. 

<img width="597" alt="TrueFans___Dashboard" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/8c83b32e-9c41-4f34-8e37-004c67c7f3f0">

We added the [Person Tag](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#person) e.g to describe the person organising the event and a [Location Tag](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#location).  We included an image tag as the event coverart maybe different to the podcast art. The link field is critical as this links out to the website or ticket page of the organiser. 

**Image 4:** TrueFans backend for Podcasts to create an Event with tickets.

<img width="576" alt="TrueFans___Dashboard" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/cee22d29-3c50-49d8-8264-1c67b001332e">

Once again @nathangathright has combined all of this into one example Event.RSS feed. 

### Example
```xml
<rss xmlns:podcast="https://podcastindex.org/namespace/1.0"
  xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd"
  xmlns:content="http://purl.org/rss/1.0/modules/content/"
  xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
<channel>
  <title>Nourish Your Network: The Confidence Booster Session</title>
  <description>Live podcast recording, hosted by Lucy Wager and Amy Wilkinson with special guests.</description>
  <podcast:medium>events</podcast:medium>
  <item>
    <guid>...</guid>
    <title>Nourish Your Network</title>
    <description>From Nerves to Narratives: How to present YOUR story with Strategist and Storyteller, Mark Whalley</description>
    <pubDate>Wed, 26 Jun 2024 12:30:00 BST</pubDate>
    <podcast:location geo="geo:35.2248586,-80.8504951?z=17">Asda, MCE, Leeds, UK</podcast:location>
    <link>https://podlifeevents.com/event-details/nourish-your-network-26th-jun-tickets</link>
    <podcast:person role="Host">Amy Wilkinson</podcast:person>
    <podcast:person role="Host"> Lucy Wager</podcast:person>
    <podcast:person role="Guest">Sydnee Washington</podcast:person>
    <podcast:person role="Guest">Dante Pittman</podcast:person>
  </item>
  <item>
    <guid>...</guid>
    <title>Asheville, NC at The Orange Peel</title>
    <description>With special guests TS Madison, and Anderson Clayton, Chair of the North Carolina Democratic Party!</description>
    <pubDate>Thu, 20 Jun 2024 20:00:00 EST</pubDate>
    <podcast:location geo="geo:35.5914061,-82.5510137?z=15">The Orange Peel</podcast:location>
    <link>https://www.etix.com/ticket/p/91197887</link>
    <podcast:person role="Host">Jon Lovett</podcast:person>
    <podcast:person role="Guest">TS Madison</podcast:person>
    <podcast:person role="Guest">Anderson Clayton</podcast:person>
  </item>
</channel>
</rss>
```

### Summary: 
1. The Podcasters RSS feed would have an additional tag called <Podcast:Events> that points to an Event URL
2. The Event URL is a full formed RSS feed with Channel and Item tags.  The Item tags contain all the details of the events. 

There is a discussion to include two more fields for each item event.  An **img tag** to show the coverart for each specific event and a **status field** to show on sale or sold out.  The event organiser could update the event feed status and via podping the app would be able to change the UI of the app. e.g change the buy button to sold out. 

3. The last discussion item is about including a publisher feed in the channel so the event organiser could show other events they are producing. 

### Example: Adding a Publisher Feed to the Event Feed. 
```xml
<channel>
  <title>Nourish Your Network: The Confidence Booster Session</title>
  <description>Live podcast recording, hosted by Lucy Wager and Amy Wilkinson with special guests.</description>
 <podcast:publisher>
        <podcast:remoteItem medium="publisher" feedGuid="003af0a0-6a45-55cf-b765-68e3d349551a" feedUrl="https://podlifeevents.com/events/feed/publisher.xml"/>            
    </podcast:publisher>
  <podcast:medium>events</podcast:medium>
  <item>
```

**Comment by samsethi** [comment]


**Reply by samsethi** [reply]
@nathangathright one question this is an event.rss how would that be included in a podcast RSS feed. A podcaster wants to add link to their events so apps can read it.  

Normally you need to add a podcast:publisher and then a URL to that feed.  

In this case it would be a podcast:events feed.

**Comment by nathangathright** [comment]
I’m glad to see all the progress here! A few points worth highlighting:
1. I’m still not in love with the idea of creating a new medium for the Events Feed vs [a JSON spec](https://gist.github.com/nathangathright/4ea6a34cfa2ecca3b8e2f9c6120901e7), but if others agree, I’ll shut up about it.
2. `Enddate` and `status` don’t have existing tags in the namespace that can be appropriated. Those will need to be fully specced out.
3. Ticket price & currency **won’t** be in the Events Feed. There’s potentially a lot of hidden complexity that should be handled to the ticketing provider’s website.
4. When the event is created within TrueFans, an external link is only optional cause you’re serving as the ticketing provider. Events pulled in from a third-party Events Feed will always need a link.
5. If Events Feeds include a `<podcast:remoteItem medium="publisher"/>` tag, then the publisher feed should also include a reciprocal `<podcast:remoteItem medium="events"/>` for each podcast’s Events Feed. Apps like TrueFans could aggregate the events across all of a publisher’s shows. If two Events Feeds have listed the same event (such as an opening band and a headliner), ideally the publisher page would only display one event. (If they have matching GUIDs, maybe this isn’t a problem.)
6. I’d assume App Review won’t take issue with in-person events (see [3.1.3(e) Goods and Services Outside of the App](https://developer.apple.com/app-store/review/guidelines/#goods-and-services-outside-of-the-app)), but I expect they take issue with linking out to purchase virtual events. Requiring virtual events to include `<podcast:location>Online</podcast:location>` would enable mobile developers to treat those events differently.

**Comment by samsethi** [comment]
An extension of this **Events** proposal might include ticketing for **Online Live Events**.  The good news nothing has to change. 
The Content Link could be repurposed to include an external link to a ticketing website. What is needed next though is an on app payment so we can support V4V payments/splits.  Of course the external ticketing site could support V4V payments/splits but at the moment I assume they all only support FIAT.  


<img width="1402" alt="TrueFans___Dashboard" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/7313ae03-9b61-4236-a9c8-022eb65c27fc">

<img width="1347" alt="TrueFans___Dashboard" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/5e3b1e74-f251-4bd8-a565-aff9d7ee8570">
